src/cargo/ops/cargo_compile.rs

```diff
@@ -397,8 +397,7 @@ pub fn compile_ws<'a>(
 
         // Include features enabled for use by dependencies so targets can also use them with the
         // required-features field when deciding whether to be built or skipped.
-        let deps = resolve_with_overrides.deps(package_id);
-        for dep in deps {
+        for (dep, _) in resolve_with_overrides.deps(package_id) {
             for feature in resolve_with_overrides.features(dep) {
                 features.insert(dep.name().to_string() + "/" + feature);
```

Comment:
Should we use renamed name here as well? So that, for example, you can specify different features for different major versions of the same package?

Reply:
Hm I'm not sure, I'll log this on the tracking issue.